### ISSUE TITLE
gooddata-ruby depending on active support 4.1

### ISSUE THREAD

**Issue by durhamka** [gooddata/gooddata-ruby#503]
We are trying to upgrade to Rails 4.2 and are being blocked by the fact that this gem depends on active support 4.1. (Rails 4.2 requires active support 4.2) Is there a plan in the near future to depend on 4.2? Thanks!

**Comment by fluke777** [comment]
@durhamka 
Hi there is an ongoing discussion how to make footprint of the gem smaller. I will update it here once I know more.

**Comment by durhamka** [comment]
Awesome, thank you. I appreciate it. Do you have a work around for this in the mean time?

**Comment by fluke777** [comment]
@durhamka Workaround is to change the version of the active_support and building the gem yourself (or using bundler). The reason why we are so strict in the gem spec is that historically/currently the main customer were consultants which are not so strong in Ruby and we wanted to limit number of combinations. Chances are gem will work with older or newer AS just fine.

**Comment by fluke777** [comment]
@durhamka I just talked to our PMs. We should have a resolution and possible roadmap by the end of the week.

**Comment by durhamka** [comment]
Awesome thank you so much. We appreciate it here on our end (GoSpotCheck)

**Comment by fluke777** [comment]
@durhamka 

Hey. We discussed it internally and we planned removal of dependency for next sprint so it should be part of next release (ETA 4 weeks).

I did some investigation and we are using ~7 methods. Some of them I would like to keep on core classes like Hash#slice or Array#compact but some we will extract to helpers (deep dup) so we minimize possibility of any clashes.

Let me know if this helps.

**Comment by durhamka** [comment]
Awesome, thanks for letting us know!

**Comment by fluke777** [comment]
@durhamka 

Just FYI we got rid of the Active Support dependency and the code hit the develop branch. If you have time and would help us test it that would be awesome. We still depend on thread-safe (which is AS dependency in later version) but we do not enforce any specific one so you should be able to hopefully use any AS version. Any feedback is highly welcome.

FYI. There is also another story coming that should make the footprint even smaller so stay tuned.  (AWS dependencies and couple others should go away) so this should further simplify integrating with GD SDK.

**Comment by durhamka** [comment]
Awesome, thanks for the update! We appreciate it.
